src/xdsl/dialects/affine.py

```diff
@@ -36,3 +42,208 @@ def store(self, value: Union[Operation, SSAValue],
             SSAValue.get(i),
             SSAValue.get(j)
         ], [], {})
+
+class AffineOperation(Enum):
+    add = '+'
+    mul = '*'
+    ceildiv = '/'
+    floordiv = '//'
+    mod = '%'
+    neg = '-'
+
+@dataclass
+class AffineExpr:
+    def __add__(self, other: int):
+        return self + AffineConstantExpr(other)
+    def __add__(self, otherExpr: AffineExpr):
+        return AffineBinaryExpr(AffineOperation.add, self, otherExpr)
+    def __sub__(self, other: int):
+        return self - AffineConstantExpr(other)
+    def __sub__(self, other: AffineExpr):
+        return AffineBinaryExpr(AffineOperation.add, self, (other * -1))
+    def __mul__(self, other: int):
+        return self * AffineConstantExpr(other)
+    def __mul__(self, other: AffineExpr):
+        return AffineBinaryExpr(AffineOperation.mul, self, other)
+    # used for ceildiv
+    def __truediv__(self, other: int): 
+        return self / AffineConstantExpr(int)
+    def __truediv__(self, other: AffineExpr): 
+        return AffineBinaryExpr(AffineOperation.ceildiv, self, other)
+    def __floordiv__(self, other: int):
+        return self // AffineConstantExpr(int)
+    def __floordiv__(self, other: AffineExpr):
+        return AffineBinaryExpr(AffineOperation.floordiv, self, other)
+    def __mod__(self, other: int):
+        return self % AffineConstantExpr(int)
+    def __mod__(self, other: AffineExpr):
+        return AffineBinaryExpr(AffineOperation.mod, self, other)       
+    def __neg__(self):
+        return self * -1
+        
+    # yapf: disable
+    def parse(parser: Parser) -> AffineExpr:
+        def get_affine_binary_expr(op: AffineOperation, lhs: AffineExpr, rhs: AffineExpr) -> AffineBinaryExpr:
+            match op:
```

Comment:
I think I would prefer now if we stay with 3.8 compatibility, since 3.9 and onwards is not that widely used yet.